src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -225,13 +228,27 @@ void Complete ()
 			}
 		}
 
+		static bool TypeHasDynamicInterfaceCastableImplementationAttribute (TypeDefinition type)
+		{
+			foreach (var ca in type.CustomAttributes) {
+				TypeDefinition caType = ca.AttributeType.Resolve ();
+				if (caType.Name == "DynamicInterfaceCastableImplementationAttribute" && caType.Namespace == "System.Runtime.InteropServices")
```

Comment:
I think this is a linker attribute only right? Should we remove it during trimming automatically?

Reply:
This attribute actually isn't linker only. The runtime enforces that you use this attribute on the implementation to ensure that your implementation will be linker-friendly.

Reply:
That sounds odd and presumably, this is only supported by RyuJIT, right? Why is there no analyzer for this instead?

Reply:
> That sounds odd and presumably, this is only supported by RyuJIT, right? Why is there no analyzer for this instead?

If the runtime didn't enforce this attribute, people wouldn't put the attribute here, illink would strip stuff, and people get weird failure modes at runtime with trimming enabled without a warning (unless we warn for all `IDynamicInterfaceCastable` uses and basically declare the whole feature not trimming safe).

Runtime enforces this attribute so that we give illink a chance to produce a consistent behavior (if the attribute is not there, illink is free to remove stuff since runtime is going to reject this either way).